src/Cli/ConfigGenerator.cs

```diff
@@ -1091,6 +1082,64 @@ public static bool TryStartEngineWithOptions(StartOptions options, FileSystemRun
             return Azure.DataApiBuilder.Service.Program.StartEngine(args.ToArray());
         }
 
+        /// <summary>
+        /// Runs all the validations on the config file and returns true if the config is valid.
+        /// </summary>
+        public static bool IsConfigValid(ValidateOptions options, FileSystemRuntimeConfigLoader loader, IFileSystem fileSystem)
+        {
+            if (!TryGetFinalConfigForRuntimeEngine(options.Config, loader, fileSystem, out string runtimeConfigFile))
+            {
+                return false;
+            }
+
+            // Validates that config file has data and it is properly deserialized
+            // Replaces all the environment variables while deserializing when starting DAB.
+            if (!loader.TryLoadKnownConfig(out RuntimeConfig? deserializedRuntimeConfig, replaceEnvVar: true))
+            {
+                _logger.LogError("Failed to parse the config file: {runtimeConfigFile}.", runtimeConfigFile);
+                return false;
+            }
+            else
+            {
+                _logger.LogInformation("Loaded config file: {runtimeConfigFile}", runtimeConfigFile);
+            }
+
+            RuntimeConfigProvider runtimeConfigProvider = new(loader);
+
+            ILoggerFactory loggerFactory = Utils.GetLoggerFactoryForCli();
+            ILogger<RuntimeConfigValidator> runtimeConfigValidatorLogger = loggerFactory.CreateLogger<RuntimeConfigValidator>();
```

Comment:
You alrady create a logger factory in `Program.cs` in the CLI project. Why another factory? now loggers are being created from different(multiple) factories?

Reply:
created a static variable to not create multiple new instances of loggerFactory